### PR TITLE
BUILD/IB: added content of ucx-ib.pc module file

### PR DESCRIPTION
**Pull request by hoopoepg**
- added content of .pc file for IB module
- updated CI scripts to test IB components

### CODE REVIEW

**Comment by avildema** [comment]
@hoopoepg Please add 'glibc-static' to buildlib/dockers/centos7.Dockerfile 
https://github.com/openucx/ucx/blob/master/buildlib/dockers/centos7.Dockerfile#L30

**Comment by hoopoepg** [comment]
> @hoopoepg Please add 'glibc-static' to buildlib/dockers/centos7.Dockerfile
> https://github.com/openucx/ucx/blob/master/buildlib/dockers/centos7.Dockerfile#L30

done

**Review by avildema — DISMISSED** [review]

File: buildlib/pr/main.yml [hunk 1]
```diff
@@ -1,12 +1,16 @@
 variables:
   DOCKER_OPT_VOLUMES: -v /hpc/local:/hpc/local -v /auto/sw_tools:/auto/sw_tools
   DOCKER_OPT_GPU: --gpus all --ulimit memlock=-1:-1 --device=/dev/infiniband/
+  DOCKER_OPT_IB: --ulimit memlock=-1:-1 --device=/dev/infiniband/
```
Comment (yosefe) [inline buildlib/pr/main.yml:4]: can we unite these to DOCKER_OPT_IB_GPU ?

**Comment by hoopoepg** [comment]
@yosefe ok to squash?

File: buildlib/pr/main.yml [hunk 2]
```diff
@@ -173,16 +177,25 @@ stages:
...
             rhel82:
               CONTAINER: rhel82
             fedora34:
               CONTAINER: fedora34
               long_test: yes
+            centos7:
+              CONTAINER: centos7_ib
+              test_static: yes
+              extra_modules: ucx-ib
+              extra_tls: ib
```
Comment (yosefe) [inline buildlib/pr/main.yml:198]: shouldn't this be "rc_verbs rc_mlx5 ..."

Comment (hoopoepg) [reply buildlib/pr/main.yml:198]: do you mean check for transport? (now it checks for component)

Comment (hoopoepg) [reply buildlib/pr/main.yml]: ok, script will look for transport

File: buildlib/pr/main.yml [hunk 3]
```diff
@@ -1,12 +1,16 @@
 variables:
   DOCKER_OPT_VOLUMES: -v /hpc/local:/hpc/local -v /auto/sw_tools:/auto/sw_tools
-  DOCKER_OPT_GPU: --gpus all --ulimit memlock=-1:-1 --device=/dev/infiniband/
+  DOCKER_OPT_IB: --ulimit memlock=-1:-1 --device=/dev/infiniband/
+  DOCKER_OPT_GPU: --gpus all $(DOCKER_OPT_IB)
```
Comment (yosefe) [inline buildlib/pr/main.yml:4]: i think it should be called DOCKER_OPT_GPU

File: src/uct/ib/ucx-ib.pc.in [hunk 4]
```diff
@@ -4,8 +4,13 @@
...
+exec_prefix = @exec_prefix@
+libdir = @libdir@/ucx
+
 Name: @PACKAGE@-ib
 Description: Unified Communication X Library IB module
 Version: @VERSION@
 Libs:
-Libs.private:
+Libs.private: -L${libdir} -luct_ib -Wl,--undefined=uct_ib_init @IBVERBS_LDFLAGS@ @NUMA_LIBS@
+Requires.private: libibverbs
```
[the remark this answers is not in the document]
Comment (hoopoepg) [reply src/uct/ib/ucx-ib.pc.in:16]: reverted

File: buildlib/pr/main.yml (hunk 3 above)
Comment (hoopoepg) [reply buildlib/pr/main.yml:4]: sorry?

**Comment by hoopoepg** [comment]
it seems infra issue

**Comment by yosefe** [comment]
```
##[warning]Module dev/libnl cannot be loaded
##[error]Bash exited with code '1'.
Finishing: Build
```
@avildema is this the right module name?

**Comment by hoopoepg** [comment]
> ```
> ##[warning]Module dev/libnl cannot be loaded
> ##[error]Bash exited with code '1'.
> Finishing: Build
> ```
> @avildema is this the right module name?

```
$ module load dev/libnl
$
```
it works on head node, but it seems module is not propagated to CI cluster

**Comment by hoopoepg** [comment]
@yosefe ok to squash?

**Review by yosefe — DISMISSED** [review]

**Review by yosefe — APPROVED** [review]

**Pull request merged**